src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -233,6 +233,10 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleExit();
             }
 
+            // Update charts for TodoList
+            todoListPanel.updateCompletionChart(logic.getFilteredTodoList());
+            todoListPanel.updateFutureBar(logic.getFilteredTodoList());
+
```

Comment:
Yes I know, but currently the chart is not listening from the `TodoList` since it must listen from a list that contains the **statistics** about the `TodoList` (Unless know how to update it directly? Because I think we still need to process the data from the `TodoList` into another list that will be represented in the chart) 

In this case what do you think if add another list in `ModelManager` class that specifically stores the statistics about the `TodoList`. So updating the statistics will be done in each of the command class i.e. same as updating the `TodoList`?